frontend/src/libs/components/meditation-player/components/audio-player/progress-bar/progress-bar.tsx

```diff
@@ -0,0 +1,49 @@
+import { formatTime } from '#libs/helpers/helpers.js';
+import { useCallback } from '#libs/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  progressBarReference: React.RefObject<HTMLInputElement | null>;
+  audioReference: React.RefObject<HTMLAudioElement | null>;
+  duration: number;
+  timeProgress: number;
+};
+
+const ProgressBar: React.FC<Properties> = ({
+  progressBarReference,
+  audioReference,
+  duration,
+  timeProgress,
+}) => {
+  const handleProgressChange: React.ChangeEventHandler<HTMLInputElement> =
+    useCallback(
+      (event) => {
+        if (audioReference.current && progressBarReference.current) {
+          const newTime = Number.parseFloat(event.target.value);
+          if (!Number.isNaN(newTime)) {
+            audioReference.current.currentTime = newTime;
+          }
+        }
+      },
+      [audioReference, progressBarReference],
+    );
+
+  return (
+    <div>
+      <div className={styles['wrapper']}>
+        <span className={styles['time']}>{formatTime(timeProgress)}</span>
+        <span className={styles['time']}>{formatTime(duration)}</span>
+      </div>
+      <input
+        className={styles['progress']}
+        type="range"
+        ref={progressBarReference as React.RefObject<HTMLInputElement>}
```

Comment:
You are using type assertion on nullable generic type. If in that case it will definitely not be null, I think it's better to write as in props in parent component

Reply:
I think we also can wrap this component into `forwardRef` and then simply use the `ref` prop in the parent component